### PR TITLE
feat(build): publish types as part of `@ember/test-helpers`

### PR DESCRIPTION
**Pull request by buschtoens**
**Build and publish the TypeScript `.d.ts` files as part of `@ember/test-helpers` to obsolete `@types/ember__test-helpers`.**

- https://github.com/emberjs/rfcs/pull/800
- https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59014
- #596 _fixed by_
  https://github.com/documentationjs/documentation/pull/1484

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
     "@typescript-eslint/parser": "^4.29.0",
     "broccoli-babel-preset-typescript": "^1.0.1",
     "broccoli-merge-trees": "^4.2.0",
-    "documentation": "^13.2.5",
+    "documentation": "npm:@buschtoens/documentation@13.2.6",
```
Comment (buschtoens) [inline package.json:64]: We may want to consider taking this opportunity to switch to a more well-maintained tool, like [API Extractor](https://api-extractor.com/). That would also improve the `.d.ts` build process.

**Comment by chriskrycho** [comment]
I just remembered this existed! 😓 @buschtoens if you can rebase and update, we'll get it published.

**Review by chriskrycho — CHANGES_REQUESTED** [review]
Had a couple comments in flight. I'm sure there will be more.

**Comment by buschtoens** [comment]
@chriskrycho Sure! I'll try to get to it in the next hours. Thanks for picking this up again. 🥳

File: package.json (hunk 1 above)
Comment (buschtoens) [reply package.json:64]: @chriskrycho

- Should I keep pointing at my fork here?
- Or should we switch to a different tool like [API Extractor](https://api-extractor.com/) first? If so, ...
  - Which tool would prefer?
  - Should this be split out into an extra PR that is merged before this one?
- Or do you have a completely different solution to this in mind?

**Comment by chriskrycho** [comment]
So sorry I let it sit for so long! Just fell off my radar, as these things do!

**Comment by chriskrycho** [comment]
@buschtoens also relevant: #1234, which… may… supersede this? (I had obviously *really* forgotten about this PR. 😢)

File: package.json (hunk 1 above)
Comment (chriskrycho) [reply package.json:64]: I think you're exactly right directionally! Let's see if we can get [TypeDoc](https://typedoc.org) wired up to do what we want. TypeDoc is almost certainly what we're going to end up using for Ember itself; it is how we are doing *all* of our TS-powered docs-generation at LinkedIn; and it is what I've used most successfully on other TS projects.

(API Extractor is (a) barely maintained at this point and (b) uninterested in supporting anything other than single-entry-point Node-style packages. TypeDoc is well-maintained, open to contributions, and actively working on supporting things like package `exports` maps. If we find gaps, let's help fix them there!)

**Comment by chriskrycho** [comment]
Going to go ahead and close this as superseded by a number of other commits; thanks again for getting it started @buschtoens!

**Pull request closed without merging**